src/content/docs/merge-queue/monorepo.mdx

```diff
@@ -0,0 +1,324 @@
+  if: ${{ needs.scopes.outputs.merge-queue == 'true' }}
+  needs: scopes
+  runs-on: ubuntu-22.04
+  steps:
+    - uses: actions/checkout@v5
+    - name: Run expensive integration tests
+      run: npm run test:integration
+```
+
+## Important Behaviors
+
+### Scope Detection is PR-Specific
+
+The `gha-mergify-ci` action only analyzes files changed by the specific pull request, **not** files
+from other PRs in the merge queue batch. This ensures:
+
+- Each PR's scopes reflect only its own changes
+- Batching decisions remain consistent even as the queue changes
+- Tests run for the correct scopes regardless of what else is in the batch
+
+### Path Filtering vs Scopes
+
+GitHub Actions offers path filtering (`on.pull_request.paths`), but it has critical limitations in
+merge queue scenarios:
+
+```yaml
+# ❌ Don't use path filtering for merge queues
+on:
+  pull_request:
+    paths:
+      - 'api/**'
+```
+
+**Problems with path filtering:**
+
+- When a job doesn't run, you can't distinguish between "filtered out" and "CI failed to start"
+
+- Required status checks fail if jobs are skipped due to filtering
+
+- In merge queues, you don't want to skip tests on PR2 just because PR1 in the batch modified
+  different files
+
+**✅ Use scopes instead:**
+
+- Jobs always run but can conditionally skip work based on scope detection
+- Status checks always report (success or skipped)
+- Merge queue batching respects scope boundaries
+
+## Example: Multi-Language Monorepo
+
+Here's a real-world example for a monorepo with Python, JavaScript, and Go services:
+
+```yaml
+scopes:
+  source:
+    files:
+      python-api:
+        includes:
+          - services/api/**/*.py
+          - libs/python/**/*.py
+      user-service:
+        includes:
+          - services/users/**/*.go
+      frontend:
+        includes:
+          - apps/web/**/*.{js,jsx,ts,tsx}
+      shared-config:
+        includes:
+          - config/**/*
+          - docker/**/*
+
+queue_rules:
+  - name: default
+    batch_size: 8
+    batch_max_wait_time: 5 min
+    queue_conditions:
+      - check-success=alls-green
+    merge_conditions:
+      - check-success=alls-green
+```
+
+With this configuration:
+- PRs affecting only `frontend` will batch together
+
+- PRs affecting `python-api` will batch together
+
+- PRs affecting `shared-config` will batch with everything (since config affects all
+  services)
+
+- Independent service changes can be tested in parallel batches
+
+## Best Practices
+
+### 1. Define Clear Scope Boundaries
+
+```yaml
+# ✅ Good: Clear, non-overlapping scopes
+scopes:
+  source:
+    files:
+      backend:
+        includes:
+          - backend/**/*
+      frontend:
+        includes:
+          - frontend/**/*
+```
+
+### 2. Handle Shared Dependencies
+
+If you have shared libraries, consider whether changes should affect all dependent
+scopes:
+
+```yaml
+# Shared lib changes affect everything
+shared-libs:
+  includes:
+    - libs/common/**/*
```

Comment:
This seems out of context so I have no clue what it means here.

Reply:
I moved it into the above paragraph it's clearer